### PR TITLE
Support full file reporting on Phabricator

### LINKED ISSUE #230: Negative line in issues crash the bot
As seen [here](https://firefoxci.taskcluster-artifacts.net/DdmbgzRTSIaqXW_jUdIGkQ/0/public/logs/live_backing.log)

### PR DESCRIPTION
**Pull request by La0**
Fixes #230

### CODE REVIEW

File: bot/code_review_bot/report/base.py [hunk 1]
```diff
@@ -30,6 +31,7 @@
         "defect": " - {nb} found by mozlint",
         "analyzer": " - `./mach lint --warnings path/to/file` (JS/Python/etc)",
     },
+    DefaultIssue: {"defect": " - {nb} found by a generic analyzer"},
```
Comment (marco-c) [inline bot/code_review_bot/report/base.py]: Could we use the name of the analyzer here?

File: bot/code_review_bot/report/phabricator.py [hunk 2]
```diff
@@ -191,7 +191,9 @@ def comment_inline(self, revision, issue, existing_comments=[]):
         comment = {
             "diffID": revision.diff_id,
             "filePath": issue.path,
-            "lineNumber": issue.line,
+            "lineNumber": issue.line
+            if issue.line is not None
+            else 1,  # support full file
```
Comment (marco-c) [inline bot/code_review_bot/report/phabricator.py]: Did you try this out on a sample Phabricator diff? Could you share the link? (I'm curious to see what it looks like, does it expand the whole file to show the comment?)

Comment (La0) [reply bot/code_review_bot/report/phabricator.py]: I tried with 0 [here](https://phabricator.services.mozilla.com/D52765#1611466) but Phabricator did not display the inline comment (still acepting it). So i'm using 1, so it should show up on the first line

File: bot/code_review_bot/report/base.py (hunk 1 above)
Comment (La0) [reply bot/code_review_bot/report/base.py]: In a followup yes (under #233), as it involves modifying the stats calculation and how we refer to analyzers.

File: bot/code_review_bot/report/phabricator.py (hunk 2 above)
Comment (La0) [reply bot/code_review_bot/report/phabricator.py]: Here is a sample output with current code: https://phabricator-dev.allizom.org/D1535#inline-20255

**Comment by La0** [comment]
And a test on a file not expanded: https://phabricator-dev.allizom.org/D1534#inline-20267

**Pull request merged**